BinaryTreeLevelOrderTraversal.java

```diff
@@ -0,0 +1,70 @@
+import java.io.*;
+import java.util.*;
+
+public class TreeNode {
+    int val;
+    TreeNode left;
+    TreeNode right;
+    TreeNode() {}
+    TreeNode(int val) { this.val = val; }
+    TreeNode(int val, TreeNode left, TreeNode right) {
+        this.val = val;
+        this.left = left;
+        this.right = right;
+    }
+}
+
+class BinaryTreeLevelOrderTraversal {
+    public List<List<Integer>> levelOrder(TreeNode root) {
+        // here the idea is to use bfs approach as we are going level by level
+
+        List<List<Integer>> levels = new ArrayList<List<Integer>>();
+        if (root == null) return levels;
+
+        Queue<TreeNode> queue = new LinkedList<TreeNode>();
+        queue.add(root);
+        int level = 0;

```

Comment:
Very minor thing: `level` is a cool variable to have for clarity, but it's not technically necessary.  The following does the same thing:

```
while (!queue.isEmpty()) {
    ArrayList<Integer> thisLevel = new ArrayList<Integer>();
    levels.add(thisLevel);

    // ...
}
```